From 9b89f58caec20bab00901264a8fe44873cf02948 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian@centricular.com>
Date: Fri, 25 Nov 2016 23:47:15 +0200
Subject: [PATCH 10/15] ext: Fix various compiler warnings with clang

wldisplay.c:179:15: error: comparison of unsigned enum expression < 0 is always false [-Werror,-Wtautological-compare]
  if (shm_fmt < 0)
      ~~~~~~~ ^ ~

gstsegmentation.cpp:419:40: error: implicit conversion from 'int' to 'char' changes value from 255 to -1 [-Werror,-Wconstant-conversion]
          filter->cvFG->imageData[j] = 255;
                                     ~ ^~~

https://bugzilla.gnome.org/show_bug.cgi?id=775112
---
 ext/wayland/wldisplay.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/wayland/wldisplay.c b/ext/wayland/wldisplay.c
index 07a203c..39782bc 100644
--- a/ext/wayland/wldisplay.c
+++ b/ext/wayland/wldisplay.c
@@ -176,7 +176,7 @@ gst_wl_display_check_format_for_shm (GstWlDisplay * display,
   guint i;
 
   shm_fmt = gst_video_format_to_wl_shm_format (format);
-  if (shm_fmt < 0)
+  if (shm_fmt == (enum wl_shm_format) -1)
     return FALSE;
 
   formats = display->shm_formats;
-- 
2.7.4

